tests/test_fastavro.py

```diff
@@ -1898,3 +1899,77 @@ def test_more_null_union_issues():
     ]
 
     assert expected == roundtrip(schema, records)
+
+
+def test_logical_type_in_union():
+    schema = {
+        "type": "record",
+        "name": "test_logical_type_in_union",
+        "fields": [{
+            "name": "item",
+            "type": [
+                "null",
+                {
+                    "name": "customLogicalType",
+                    "type": "int",
+                    "logicalType": "date"
+                }
+            ]
+        }]
+    }
+
+    records = [
+        {"item": None},
+        {"item": "2019-05-06"}
+    ]
+
+    expected = [
+        {"item": None},
+        {"item": datetime.date(2019, 5, 6)}
+    ]
+
+    assert expected == roundtrip(schema, records)
+
+
+def test_named_schema_with_logical_type_in_union():
+    named_schema = {
+        "name": "named_schema_with_logical_type",
+        "namespace": "com.example",
+        "type": "record",
+        "fields": [
+            {
+                "name": "item",
+                "type": {
+                    "name": "customLogicalType",
+                    "type": "int",
+                    "logicalType": "date"
+                }
+            }
+        ]
+    }
+
+    fastavro.parse_schema(named_schema)
+
+    schema = {
+        "type": "record",
+        "name": "test_named_schema_with_logical_type",
+        "fields": [{
+            "name": "item",
+            "type": [
+                "null",
+                "com.example.named_schema_with_logical_type"
+            ]
+        }]
+    }
+
+    records = [
+        {"item": None},
+        {"item": {"item": "2019-05-06"}}
+    ]
+
+    expected = [
+        {"item": None},
+        {"item": {"item": datetime.date(2019, 5, 6)}}
+    ]
+
+    assert expected == roundtrip(schema, records)
```

Comment:
I'm not sure if I placed the tests correctly.

Reply:
I think this is an okay place for them. They could possibly go under `test_validation` but there's not a strong reason to move them there.